airflow/api_fastapi/core_api/routes/public/pools.py

```diff
@@ -167,15 +168,57 @@ def patch_pool(
 @pools_router.post(
     "/",
     status_code=status.HTTP_201_CREATED,
-    responses=create_openapi_http_exception_doc([status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN]),
+    responses=create_openapi_http_exception_doc(
+        [status.HTTP_401_UNAUTHORIZED, status.HTTP_403_FORBIDDEN, status.HTTP_409_CONFLICT]
+    ),
 )
 def post_pool(
     post_body: PoolPostBody,
     session: Annotated[Session, Depends(get_session)],
 ) -> PoolResponse:
     """Create a Pool."""
+    pool = session.scalar(select(Pool).where(Pool.pool == post_body.pool))
+    if pool is not None:
+        raise HTTPException(status.HTTP_409_CONFLICT, f"Pool with name: `{post_body.pool}` already exists")
```

Comment:
Database will handle this for us. We do not need an extra query and checking in the application code. (Just catch the duplicate entry exception from the DB)

Reply:
I noticed that the API handling the creation of connections includes an additional database query to check for duplicates. Reference: https://github.com/apache/airflow/blob/main/airflow/api_fastapi/core_api/routes/public/connections.py#L136-#L138.

Should we adopt a consistent approach across the codebase by either relying on handling database insert exceptions or using additional database queries?

Reply:
Yes we should. 

I think relying on the DB to implement and do the unique constraint checks is easier (and more resilient). We just need to catch the appropriate sqlalchemy error.

Reply:
Yes, there is a follow-up story to make them consistent. I have noted to include these parts, too. Let's use the opportunity to make it consistent and efficient.